junit-jupiter-api/src/main/java/org/junit/jupiter/api/parallel/ResourceLock.java

```diff
@@ -40,6 +40,9 @@
  * <p>Since JUnit Jupiter 5.4, this annotation is {@linkplain Inherited inherited}
  * within class hierarchies.
  *
+ * <p>Since JUnit Jupiter 5.10, this annotation can be used to specify the target
+ * of the lock with {@link ResourceLockTarget}.
+ *
```

Comment:
What happens if a class has a `@ResourceLock` annotation with target `CHILDREN` and a method has one for the same resource? Does the one from the method override the one from the class? This should be documented here.